From 1a3d2da9cd62573555f3eaeacbd47b4823933644 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:09 -0400
Subject: [tools] lib/traceevent: Get rid of die() in create_arg_item()

Message-id: <1408953210-26343-122-git-send-email-jolsa@redhat.com>
Patchwork-id: 88915
O-Subject: [PATCH RHEL7.1 BZ1133083 121/322] tools lib traceevent: Get rid of die() in create_arg_item()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 2036fcd1c7ce455424c11bdb1c8a2ac906430e2f
Author: Namhyung Kim <namhyung@kernel.org>
Date: Mon Dec 9 14:34:05 2013 +0900
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/parse-filter.c b/tools/lib/traceevent/parse-filter.c
index 2b73abf..53e48eb 100644
--- a/tools/lib/traceevent/parse-filter.c
+++ b/tools/lib/traceevent/parse-filter.c
@@ -362,8 +362,11 @@ create_arg_item(struct event_format *event, const char *token,
   arg->value.type =
    type == EVENT_DQUOTE ? FILTER_STRING : FILTER_CHAR;
   arg->value.str = strdup(token);
-  if (!arg->value.str)
-   die("malloc string");
+  if (!arg->value.str) {
+   free_arg(arg);
+   show_error(error_str, "failed to allocate string filter arg");
+   return NULL;
+  }
   break;
  case EVENT_ITEM:
   /* if it is a number, then convert it */
-- 
1.7.1